backend/src/services/jobs/detect-private-hosts.ts

```diff
@@ -0,0 +1,47 @@
+import { AppDataSource } from "data-source"
+import { Hosts, ApiEndpoint } from "models"
+import { getQB, insertValuesBuilder } from "services/database/utils"
+import { MetloContext } from "types"
+import axios from "axios"
+import mlog from "logger"
+
+export const detectPrivateHosts = async (
+  ctx: MetloContext,
+): Promise<boolean> => {
+  let queryRunner = AppDataSource.createQueryRunner()
+  try {
+    await queryRunner.connect()
+    const hosts = await getQB(ctx, queryRunner)
+      .select(["host"])
+      .from(ApiEndpoint, "endpoint")
+      .distinct(true)
+      .groupBy("host")
+      .getRawMany()
+    const vals = await Promise.all(
+      hosts.map(async ({ host }) => {
+        let isPublic = false
+        try {
+          const resp = await axios.get(`http://${host}`, { timeout: 5000 })
+          if (resp && resp.status) {
+            isPublic = true
+          }
+        } catch (err) {
```

Comment:
This is only going to catch the first promise that fails, I think it's better to use Promise.allSettled